File: src/main/java/com/polytech/webscraipper/sdk/FeignConfig.java
package com.polytech.webscraipper.sdk;

import feign.Client;
import feign.RequestInterceptor;
import java.nio.charset.StandardCharsets;
import java.util.Base64;
import java.util.concurrent.TimeUnit;
import okhttp3.OkHttpClient;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class FeignConfig {

  private final String authHeader;

  public FeignConfig(
      @Value("${langfuse.api.username}") String username,
      @Value("${langfuse.api.password}") String apiKey) {
    this.authHeader =
        "Basic "
            + Base64.getEncoder()
                .encodeToString((username + ":" + apiKey).getBytes(StandardCharsets.UTF_8));
  }

  @Bean
  public RequestInterceptor authInterceptor() {
    return template -> template.header("Authorization", authHeader);
  }

  @Bean
  public Client feignClient() {
    OkHttpClient okHttpClient =
        new OkHttpClient.Builder()
            .connectTimeout(10, TimeUnit.SECONDS)
            .readTimeout(30, TimeUnit.SECONDS)
            .writeTimeout(30, TimeUnit.SECONDS)
            .followRedirects(true)
            .build();
    return new feign.okhttp.OkHttpClient(okHttpClient);
  }
}

File: src/main/java/com/polytech/webscraipper/sdk/FeignConfig.kt
package com.polytech.webscraipper.sdk

import feign.Client
import feign.RequestInterceptor
import okhttp3.OkHttpClient
import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import java.util.*
import java.util.concurrent.TimeUnit

@Configuration
open class FeignConfig(
    @Value("\${langfuse.api.username}")
    val username: String,
    @Value("\${langfuse.api.password}")
    val apiKey: String,
) {
    private val authHeader: String = "Basic " + Base64.getEncoder().encodeToString(("$username:$apiKey").toByteArray())

    @Bean
    open fun authInterceptor(): RequestInterceptor = RequestInterceptor { template -> template.header("Authorization", authHeader) }

    @Bean
    open fun feignClient(): Client {
        val okHttpClient = OkHttpClient.Builder()
            .connectTimeout(10, TimeUnit.SECONDS)
            .readTimeout(30, TimeUnit.SECONDS)
            .writeTimeout(30, TimeUnit.SECONDS)
            .followRedirects(true)
            .build()
        return feign.okhttp.OkHttpClient(okHttpClient)
    }
}
